include/flatbuffers/flatbuffers.h

```diff
@@ -394,6 +400,34 @@ template<typename T> class Vector {
   }
 };
 
+template<class U>
+FLATBUFFERS_CONSTEXPR_CPP11 flatbuffers::span<U> make_span(Vector<U> &vec)
+    FLATBUFFERS_NOEXCEPT {
+  static_assert(Vector<U>::is_span_observable, "wrong type U");
```

Comment:
`wrong type U` is slightly confusing if this ever gets compiled on big endian, but that should be so rare that I am fine with it :)

Reply:
Fixed.
`Vector<Offset<Table>>` is also "wrong type U", for example `flatbuffers::make_bytes_span(*monster->testarrayofstring())`.
I can add SFINE to exclude all "wrong" types from the overload resolution. But it will be ugly and will slow down compilation.

Reply:
not necessary.. was just a thought.